v2/datastream-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToBigQueryIT.java

```diff
@@ -0,0 +1,209 @@
+  private DatastreamResourceManager datastreamResourceManager;
+  private BigQueryResourceManager bigQueryResourceManager;
+
+  private static final int ROW_COUNT = 10;
+
+  private static final String ID = "id";
+  private static final String JOB = "job";
+  private static final String NAME = "name";
+
+  @Before
+  public void setUp() throws IOException {
+    pubsubResourceManager =
+        PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+    bigQueryResourceManager =
+        BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+    datastreamResourceManager =
+        DatastreamResourceManager.builder(testName, PROJECT, REGION)
+            .setCredentialsProvider(credentialsProvider)
+            .setPrivateConnectivity("datastream-private-connect-us-central1")
+            .build();
+    jdbcResourceManager = CustomMySQLResourceManager.builder(testName).build();
+  }
+
+  @After
+  public void cleanUp() {
+    ResourceManagerUtils.cleanResources(
+        jdbcResourceManager,
+        pubsubResourceManager,
+        datastreamResourceManager,
+        bigQueryResourceManager);
+  }
+
+  @Ignore("Flaky test")
+  @Test
+  public void testDataStreamToBigQuery() throws IOException {
+    baseDataStreamToBigQuery(Function.identity());
+  }
+
+  @Ignore("Flaky test")
+  @Test
+  public void testDataStreamToBigQueryUsingAtleastOnceMode() throws IOException {
+    ArrayList<String> experiments = new ArrayList<>();
+    experiments.add("streaming_mode_at_least_once");
+    baseDataStreamToBigQuery(
+        b ->
+            b.addEnvironment("additionalExperiments", experiments)
+                .addEnvironment("enableStreamingEngine", true));
+  }
+
+  private void baseDataStreamToBigQuery(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder) throws IOException {
+    // Arrange
+    // Create a BigQuery table
+    List<Field> bqSchemaFields =
+        Arrays.asList(
+            Field.of(ID, StandardSQLTypeName.INT64),
+            Field.of(JOB, StandardSQLTypeName.STRING),
+            Field.of(NAME, StandardSQLTypeName.STRING));
+    Schema bqSchema = Schema.of(bqSchemaFields);
+    bigQueryResourceManager.createDataset(REGION);
+    TableId bqTable = bigQueryResourceManager.createTable(testName, bqSchema);
+
+    // Create a Pub/Sub subscription
+    String nameSuffix = RandomStringUtils.randomAlphanumeric(8);
+    TopicName topic = pubsubResourceManager.createTopic("input-" + nameSuffix);
+    SubscriptionName subscription =
+        pubsubResourceManager.createSubscription(topic, "sub-1-" + nameSuffix);
+
+    // Create a JDBC table
+    HashMap<String, String> columns = new HashMap<>();
+    columns.put(ID, "NUMERIC NOT NULL");
+    columns.put(JOB, "VARCHAR(200)");
+    columns.put(NAME, "VARCHAR(200)");
+    JDBCResourceManager.JDBCSchema schema = new JDBCResourceManager.JDBCSchema(columns, ID);
+    String tableName = "MySqlToBigQuery";
```

Comment:
I would set this to the testName to ensure no collisions
```suggestion
    String tableName = testName;
```